nemo/collections/common/data/lhotse/cutset.py

```diff
@@ -121,61 +120,70 @@ def read_nemo_manifest(config, is_tarred: bool) -> CutSet:
-            else:
-                cuts = CutSet.mux(*[cs.repeat() for cs in cutsets], weights=weights, seed="trng")
+            cuts = CutSet(LazyNeMoIterator(config.manifest_filepath, **notar_kwargs, **common_kwargs))
     else:
+        # Format option 1:
+        #   Assume it's [[path1], [path2], ...] (same for tarred_audio_filepaths).
+        #   This is the format for multiple NeMo buckets.
+        #   Note: we set "weights" here to be proportional to the number of utterances in each data source.
+        #         this ensures that we distribute the data from each source uniformly throughout each epoch.
+        #         Setting equal weights would exhaust the shorter data sources closer the towards the beginning
+        #         of an epoch (or over-sample it in the case of infinite CutSet iteration with .repeat()).
+        # Format option 1:
+        #   Assume it's [[path1, weight1], [path2, weight2], ...] (while tarred_audio_filepaths remain unchanged).
+        #   Note: this option allows to manually set the weights for multiple datasets.
         logging.info(
-            f"Initializing Lhotse CutSet from a single NeMo manifest (non-tarred): '{config.manifest_filepath}'"
+            f"Initializing Lhotse CutSet from multiple tarred NeMo manifest sources with a weighted multiplexer. "
+            f"We found the following sources and weights: "
         )
-        cuts = CutSet(LazyNeMoIterator(config.manifest_filepath, **common_kwargs))
+        cutsets = []
+        weights = []
+        tar_paths = config.tarred_audio_filepaths if is_tarred else repeat((None,))
+        # Create a stream for each dataset.
+        for manifest_info, (tar_path,) in zip(config.manifest_filepath, tar_paths):
+            # First, convert manifest_path[+tar_path] to an iterator.
+            manifest_path = manifest_info[0]
+            if is_tarred:
+                nemo_iter = LazyNeMoTarredIterator(
+                    manifest_path=manifest_path, tar_paths=tar_path, shuffle_shards=config.shuffle, **common_kwargs
+                )
+            else:
+                nemo_iter = LazyNeMoIterator(manifest_path, **notar_kwargs, **common_kwargs)
+            # Then, determine the weight or use one provided
+            if len(manifest_info) == 1:
+                weight = len(nemo_iter)
+            else:
+                assert (
+                    isinstance(manifest_info, Sequence)
+                    and len(manifest_info) == 2
+                    and isinstance(manifest_info[1], (int, float))
+                ), (
+                    "Supported inputs types for config.manifest_filepath are: "
+                    "str | list[list[str]] | list[tuple[str, number]] "
+                    "where str is a path and number is a mixing weight (it may exceed 1.0). "
+                    f"We got: '{manifest_info}'"
+                )
+                weight = manifest_info[1]
+            logging.info(f"- {manifest_path=} {weight=}")
```

Comment:
manifest_path= ? Is this new python syntax?

Reply:
Yes, since Python 3.8. `f"{x=}"` resolves to a string `"x=<actual-x-value>"`

 https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging